ijkmedia/ijkplayer/ff_ffplay.c

```diff
@@ -320,6 +322,65 @@ static int packet_queue_get(PacketQueue *q, AVPacket *pkt, int block, int *seria
     return ret;
 }
 
+/* return delete num*/
+static int packet_queue_delete_elements_until_by_pts(PacketQueue *q, int64_t pts) {
+    MyAVPacketList *pkt = NULL;
+    int delete_count = 0;
+
+    if (q->max_pts == pts) {
+        packet_queue_flush(q);
+        return delete_count;
+    }
+
+    SDL_LockMutex(q->mutex);
+    while (1) {
+        if (q->abort_request) {
+            delete_count = -1;
+            break;
+        }
+
+        pkt = q->first_pkt;
+        if (pkt) {
+            if (pkt->pkt.pts >= pts)
+                break;
+            q->first_pkt = pkt->next;
+            if (!q->first_pkt)
+                q->last_pkt = NULL;
+            q->nb_packets--;
+            q->size -= pkt->pkt.size + sizeof(*pkt);
+            if (pkt->pkt.duration > 0)
+                q->duration -= pkt->pkt.duration;
+            pkt->next = q->recycle_pkt;
+            q->recycle_pkt = pkt;
+            av_free_packet(&pkt->pkt);
+            delete_count++;
+        }
+    }
+    SDL_UnlockMutex(q->mutex);
+    return delete_count;
+}
+
+
+/* return 0 if aborted return seek pos last key frame pts*/
+static int64_t packet_video_queue_seek(PacketQueue *q, int64_t seek_pos) {
+    MyAVPacketList *pkt = NULL;
+    int64_t key_frame_pts = 0;
+
+    if (q->nb_packets == 0)
+        return key_frame_pts;
+
+    pkt = q->first_pkt;
+    while (pkt->next) {
+        if (pkt->pkt.flags & AV_PKT_FLAG_KEY)
+            key_frame_pts = pkt->pkt.pts;
+        if (pkt->pkt.pts >= seek_pos)
+            break;
+        pkt = pkt->next;
+    }
+
+    return key_frame_pts;
```

Comment:
If all packets are deleted before a key-frame is found, player would seek to a non-key-frame.